apps/movies/src/movies.controller.ts

```diff
@@ -1,16 +1,29 @@
-import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import { Controller, Get, Post, Body, Param, UseInterceptors, UploadedFile } from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { UploaderService } from '@app/common/uploader/uploader.service';
 import { MoviesService } from './movies.service';
 import { CreateMovieDto } from './dto/create-movie.dto';
 import { plainToInstance } from 'class-transformer';
 import { TrendedMovieResponseDto } from './dto/trended-movie-response.dto';
 
 @Controller('movies')
 export class MoviesController {
-  constructor(private readonly moviesService: MoviesService) {}
+  constructor(private readonly moviesService: MoviesService, private readonly uploaderService: UploaderService) {}
 
   @Post()
-  create(@Body() createMovieDto: CreateMovieDto) {
-    return this.moviesService.create(createMovieDto);
+  @UseInterceptors(FileInterceptor('image'))
+  async create(@Body() createMovieDto: CreateMovieDto, @UploadedFile() image: Express.Multer.File) {
+    const imageUrl = await this.uploaderService.upload(image.originalname, image.buffer);
+    const movie = {
+      title: createMovieDto.title,
+      description: createMovieDto.description,
+      genre: createMovieDto.genre,
+      duration: +createMovieDto.duration,
+      releaseDate: createMovieDto.releaseDate,
+      trended: createMovieDto.trended,
+      imageUrl,    
+    }
+    return this.moviesService.create(movie);
```

Comment:
```suggestion
    return this.moviesService.create({
      title: createMovieDto.title,
      description: createMovieDto.description,
      genre: createMovieDto.genre,
      duration: +createMovieDto.duration,
      releaseDate: createMovieDto.releaseDate,
      trended: createMovieDto.trended,
      imageUrl,    
    });
```

I think you can put object directly OR rename movie object to movieDetails. It's up to you